src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1719,7 +1719,11 @@ public override async void DidMoveToParentViewController(UIViewController parent
 				{
 					NavigationRenderer navRenderer;
 					if (_navigation.TryGetTarget(out navRenderer))
+					{
+						if (!navRenderer._uiRequestedPop)
+							navRenderer._uiRequestedPop = true;
```

Comment:
One thing I noticed is that this code is called if you just pop the page off the stack vs click the back button. 

So, on the second page of this app if you add a button that calls "Navigation.popasync" this gets transitioned to true but then never transitions back to false. 

if we change to instead use DidPopItem vs ShouldPopItem does that make this all work a bit better?


```C#
[Export("navigationBar:didPopItem:")]
		[Internals.Preserve(Conditional = true)]
		internal bool DidPopItem(UINavigationBar _, UINavigationItem __)
		{
            if (_ignorePopCall) return;
            _uiRequestedPop = true;
			return true;
		}
```

Reply:
@PureWeen, In the current source, the issue is not reproducible. The issue was fixed by the changes in this  [PR](https://github.com/dotnet/maui/pull/26235), so can we close this PR or consider test case alone?

Reply:
Yea, I like the idea of just taking the test case. 

Let's just get this PR merged with the test case since the other PR didn't have one

Reply:
@PureWeen, I have removed the fix. The PR now contains only the test case. Could you please review it?